classes/stageAssignment/StageAssignment.php

```diff
@@ -17,157 +17,193 @@
  *
  * @ingroup stageAssignment
  *
- * @see StageAssignmentDAO
- *
  * @brief Basic class describing a Stage Assignment.
  */
 
 namespace PKP\stageAssignment;
 
-class StageAssignment extends \PKP\core\DataObject
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use PKP\userGroup\relationships\UserGroupStage;
+
+class StageAssignment extends Model
 {
-    //
-    // Get/set methods
-    //
+    protected $table = 'stage_assignments';
+    protected $primaryKey = 'stage_assignment_id';
+    public $timestamps = false;
+
+    protected $fillable = [
+        'submissionId', 'userGroupId', 'userId', 
+        'dateAssigned', 'recommendOnly', 'canChangeMetadata'
+    ];
+
+    // Relationships
+
     /**
-     * Set the submission ID
+     * One to many relationship with user_group_stage table => UserGroupStage Eloquent Model
      *
-     * @param int $submissionId
+     * To eagerly fill the userGroupStages Collection, the calling code should add 
+     * StageAssignment::with(['userGroupStages'])
      */
-    public function setSubmissionId($submissionId)
+    public function userGroupStages(): HasMany
     {
-        $this->setData('submissionId', $submissionId);
+        return $this->hasMany(UserGroupStage::class, 'user_group_id', 'user_group_id');
     }
 
+    // Accessors and Mutators
```

Comment:
Hmm, can you explain why do we need them here, I mean shouldn't Eloquent handle this automatically?

Reply:
After retrieving a model, let's say `$stageAssignment`, by default the properties retain their DB naming. In our case, if we want to access the property `submission_id` we could write `$stageAssignment->submission_id`. 

But in the code we want to access the properties in a camel case, namely `stageAssignment->submissionId`, rather than the snake case `$stageAssignment->submission_id`.

I have not found a way to do that without those accessor/mutator functions. But any approach that will make them unneeded is very welcome.

Reply:
Looks good, thanks!